### DISCUSSION TITLE
Feature Request - 'Like' a comment (Reactions)

### DISCUSSION THREAD

**Discussion by m-j-g** [flyerhq/flutter_chat_ui#91] [Ideas & Requests]
We talked about this previously, but didn't see a feature request for it.  This is one of the most requested chat features I get.  People have become reliant on using 'Like' on iMessage, Slack or twitter/Facebook/social media as a way to acknowledge someones point without having to actually write up a response.

**Comment by 31Carlton7** [comment]
This feature would be excellent. Is it in the works by any chance devs?

**Comment by demchenkoalex** [comment]
Hey @31Carlton7 we all (maintainers) changed jobs recently so not much is going on here, but looking at the amount of reactions here, it is something that will be prioritized. Sorry about the delay.

**Comment by 31Carlton7** [comment]
All good. Just wondering, but do you guys accept others to join the team? Not requesting to be on it but it seems like other people have submitted PR's and forks and they haven't been addressed. Do you only allow people within the team? Or do you also allow anyone to contribute?

**Comment by demchenkoalex** [comment]
Wait, what? I am always reviewing PRs. There are 2, 2 days and 7h hours now, but I am working and will review them on weekend. And the last one adds bad dependencies so it is something I can't merge now. What is not addressed in your opinion?

**Comment by 31Carlton7** [comment]
Oh my bad, I mixed up PR's and Issues. I was wondering why around 15 were still open. In all fairness I believe basically all things in this repo/package are addressed. I suppose I'm just eager for new feature implementations, because of how unique the package is lol.

**Comment by demchenkoalex** [comment]
These issues are open, because anyone can ask us for design (if needed) and implement them. They require a bit more than an hour usually, see that I always resolve critical bugs and small issues. And on our side I am fixing flutter Firebase package now and we are not yet finished aligning React Native to Flutter. When these are done we can come back to add something new, definitely replies and reactions.

**Comment by Justus-M** [comment]
IMO this and chat replies are the 2 biggest things separating this package from today's mainstream chat apps. It's already the best flutter chat library IMO, but would love to see these two implemented!

**Reply by demchenkoalex** [reply]
yeah, we are on it :)

**Comment by felixgabler** [comment]
Before we can begin implementing this, we have to decide whether we want to offer different types of reactions i.e. should we only have "Thumbs up" or a handful of preselected emojis or make it fully customizable (like WhatsApp maybe?)?

I'd be in favour of selecting a handful of icons/emojis as a default and having the option of overriding the reactions icons. Not sure how we can make it fully customizable though, as we'd need to also implement an emoji keyboard in this case. Also, this could be added later if necessary. What do you think? 

Once we have enough signal, we will need to get some designs for the reaction selection popup, the reactions badges on a message as well as maybe for a list of reactions (might want to leave this for the users to implement this though i.e. just provide a builder option that gets the list of reactions by user for each message).

**Reply by m-j-g** [reply]
> 

Sounds like a cool idea.  I agree with you on having a default small number of reactions.  I would think something like ThumbsUp/ThumbsDown/Laugh would fill the minimum urge when tapping on a message to react.  But whatever yall decide I think this is going to be an awesome user experience enhancement!

**Reply by Justus-M** [reply]
People mostly use the preselected emojis from what I've seen.

Since this feature is not yet implemented, I think it would make sense to start with preselected and add the custom ones later as you suggested.

IMO simple preselected emojis and chat replies will add a ton of value for users, and customizable emojis will only add a very small amount of additional value.

**Reply by demchenkoalex** [reply]
> I'd be in favour of selecting a handful of icons/emojis as a default and having the option of overriding the reactions icons

totally agree, that what I wanted

> Not sure how we can make it fully customizable though, as we'd need to also implement an emoji keyboard in this case.

no keyboard please :D I think just a popup on a long press or something with a handful of emojis. Asked designer to implement, after typing indicator and unread messages thing

**Comment by baptistesx** [comment]
Hey guys, any news on this nice enhancement? :)
Maybe i can help a bit (i'm still not a flutter expert)

**Reply by demchenkoalex** [reply]
Not even started. Will require some architectural decisions to be made first. How to store the reactions? Is is a finite list? etc

**Comment by pinktoadette** [comment]
Is there anyway to make this as generic as possible? I’m not interested in likes, but would like my own button. There’s so way I can add my own button outside of the bubble without rebuilding the entire bubble builder.

**Reply by demchenkoalex** [reply]
how it would look like, your own button? can you do a mock up?

**Reply by pinktoadette** [reply]
![Screenshot_20230417_142041](https://user-images.githubusercontent.com/5497684/232579794-74904434-3483-4bca-af78-0f6e42a72a16.png)

Similar to Facebook like button, perhaps users can pass in some metadata via the message metadata of what they need to display, or define their own class. I think this offers much more flexibility than a strict, thumbs up/like, etc.

**Reply by demchenkoalex** [reply]
there was never an intention to have a thumbs up only, rather a long press with an array of options. This row of buttons will be something custom I suppose

**Reply by pinktoadette** [reply]
Right, the thumbs up, I was only referring to one of the responses. But I think this row offers better flexibility for more people, like the custom bottom list widget.
